public/app/core/components/AppChrome/QuickAdd/QuickAdd.tsx

```diff
@@ -0,0 +1,76 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Menu, Dropdown, Button, Icon, useStyles2, useTheme2, ToolbarButton } from '@grafana/ui';
+import { useMediaQueryChange } from 'app/core/hooks/useMediaQueryChange';
+import { useSelector } from 'app/types';
+
+import { NavToolbarSeparator } from '../NavToolbarSeparator';
+
+import { findCreateActions } from './utils';
+
+export interface Props {}
+
+export const QuickAdd = ({}: Props) => {
+  const styles = useStyles2(getStyles);
+  const theme = useTheme2();
+  const navBarTree = useSelector((state) => state.navBarTree);
+  const breakpoint = theme.breakpoints.values.sm;
+
+  const [isSmallScreen, setIsSmallScreen] = useState(window.matchMedia(`(max-width: ${breakpoint}px)`).matches);
+  const createActions = findCreateActions(navBarTree);
```

Comment:
As this is a recursive function could it make sense to wrap it in a `useMemo()`? (I am generally against any premature optimisation, but this looks like a potential candidate.)